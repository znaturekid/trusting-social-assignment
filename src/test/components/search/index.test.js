import React from 'react'
import Enzyme, {
    mount,
    shallow
} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SearchComponent from '../../../components/search'
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

Enzyme.configure({
    adapter: new Adapter()
})

function setup() {
    const props = {
        search: jest.fn(),
        query: 'singapore',
        loading: false
    }

    const component = <SearchComponent { ...props} />

    return {
        props,
        component
    }
}
jest.useFakeTimers();

describe('components', () => {
    describe('Search =>', () => {
        it('should render self and subcomponents', () => {
            const { component } = setup()
            expect(shallow(component)).toMatchSnapshot();
        })

        it('should render input with default value ', () => {
            const { component, props } = setup()
            const search = mount(component);
            const input = search.find('input');
            expect(input).toHaveLength(1);
            expect(search.state().query).toBe(props.query);
        })
        it('should call search function after change event fire', () => {
            const { component, props } = setup()
            const textChange = "vietname";
            const search = mount(component);
            const input = search.find('input');
            input.simulate('focus');
            input.simulate('change', {target: { value: textChange }});
            expect(setTimeout).toHaveBeenCalledTimes(1);
            expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 400);
            jest.runOnlyPendingTimers();
            expect(props.search).toBeCalledWith(textChange);
            input.simulate('change', {target: { value: '' }});
        })
        it('should call search function after keypress event fire with enter key', () => {
            const { component, props } = setup()
            const search = mount(component);
            const input = search.find('input');
            input.simulate('focus');
            input.simulate('keypress', {charCode: 13});
            expect(props.search).toBeCalledWith(props.query);
        })
        it('should not call search function after keypress event fire with any except enter key', () => {
            const { component, props } = setup()
            const search = mount(component);
            const input = search.find('input');
            input.simulate('focus');
            input.simulate('keypress', {charCode: 2});
            expect(props.search).not.toBeCalled();
            input.simulate('change', {target: { value: 'ab' }});
            input.simulate('keypress', {charCode: 13});
        })
    })
})