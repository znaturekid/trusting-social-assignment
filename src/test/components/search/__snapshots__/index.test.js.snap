// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Search => should render self and subcomponents 1`] = `
<styled.div
  data-loading={false}
>
  <styled.input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter text to search"
    type="text"
    value="singapore"
  />
</styled.div>
`;

exports[`components Search= should render self and subcomponents 1`] = `
<styled.div
  data-loading={false}
>
  <styled.input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter text to search"
    type="text"
    value="singapore"
  />
</styled.div>
`;

exports[`components Search= should render self and subcomponents 2`] = `
<styled.div
  data-loading={false}
>
  <styled.input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter text to search"
    type="text"
    value="singapore"
  />
</styled.div>
`;

exports[`components Search= should render self and subcomponents 3`] = `null`;
